src/CardListItem/CardListItem.js

```diff
@@ -2,49 +2,40 @@ import React, { Component } from "react";
 import "./CardListItem.css";
 
 class CardListItem extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      count: props.userCardsData.find(card => {
-        return card.cardName === props.cardName;
-      }).cardCount
-    };
+  removeListItem = () => {
+    const [...newUserCardsData] = this.props.userCardsData;
+    newUserCardsData.splice(this.props.cardIndex, 1);
+    this.props.saveArray('userCardsData', newUserCardsData);
   }
 
-  removeListItem = event => {
-    this.props.removeListItem(parseInt(event.target.closest("li").id));
```

Comment:
I realized using `event.target.closest("li").id` was unnecessary because it is returning the cardIndex, and the cardIndex can be accessed with `this.props.cardIndex` instead.